lib/logstash/inputs/jms.rb

```diff
@@ -132,10 +152,26 @@ class LogStash::Inputs::Jms < LogStash::Inputs::Threadable
   # For some known examples, see: [Example jms.yml](https://github.com/reidmorrison/jruby-jms/blob/master/examples/jms.yml)
 
   public
+
+  def initialize(*params)
+    super
+
+    unless original_params.include?('headers_target')
+      @headers_target = ecs_select[disabled: nil, v1: '[@metadata][input][jms][headers]']
```

Comment:
alternate (I'm not sure which I like better):

```suggestion
    unless original_params.include?('headers_target') || ecs_compatibility == :disabled
      @headers_target = '[@metadata][input][jms][headers]'
```

Reply:
I don't like using `||` or `&&` with unless - makes me need to stop for a second or two reading the code
but yeah the intent would be cleaner having just the one condition ... here.